app/src/main/java/net/gsantner/markor/activity/ActionOrderActivity.java

```diff
@@ -0,0 +1,235 @@
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.action_reorder_activity);
+
+        //  Set back button
+        Toolbar _toolbar = findViewById(R.id.action_reorder_toolbar);
+        setSupportActionBar(_toolbar);
+        ActionBar bar = getSupportActionBar();
+        assert bar != null;
+        bar.setDisplayHomeAsUpEnabled(true);
+
+        // Set up recyclerview
+        RecyclerView recycler = findViewById(R.id.action_reorder_recycler);
+        recycler.setLayoutManager(new LinearLayoutManager(this));
+
+        extractActionData();
+        _adapter = new Adapter(_actions);
+
+        ItemTouchHelper.Callback callback = new ReorderCallback(_adapter);
+        ItemTouchHelper helper = new ItemTouchHelper(callback);
+        helper.attachToRecyclerView(recycler);
+
+        recycler.setHasFixedSize(true);
+        recycler.setAdapter(_adapter);
+    }
+
+    @Override
+    public boolean onCreateOptionsMenu(Menu menu) {
+
+        MenuInflater inflater = getMenuInflater();
+        inflater.inflate(R.menu.reorder_actions_actionbar__menu, menu);
+
+        ContextUtils cu = ContextUtils.get();
+        cu.tintMenuItems(menu, true, Color.WHITE);
+
+        return true;
+    }
+
+    @Override
+    public boolean onOptionsItemSelected(MenuItem item) {
+        int id = item.getItemId();
+        switch (id) {
+
+            case android.R.id.home:
+                // Return to settings
+                finish();
+                return true;
+
+            case R.id.action_reorder_accept:
+                ArrayList<String> reorderedKeys = new ArrayList<>();
+                for (int i : _adapter.order) reorderedKeys.add(_keys.get(i));
+                _textActions.saveActionOrder(reorderedKeys);
+                finish();
+                return true;
+
+            default:
+                return super.onOptionsItemSelected(item);
+        }
+    }
+
+    private void extractActionData() {
+
+        // Extract actions
+        int documentType = getIntent().getExtras().getInt("key");
+
+        switch (documentType) {
+            default:
+            case R.id.action_format_markdown:
+                _textActions = new MarkdownTextActions(this, null);
+                break;
+            case R.id.action_format_todotxt:
+                _textActions = new TodoTxtTextActions(this, null);
+                break;
+            case R.id.action_format_plaintext:
+                _textActions = new PlaintextTextActions(this, null);
+                break;
+        }
+
+        Map<String, ActionItem> actionMap = _textActions.getActiveActionMap();
+        _keys = new ArrayList<>(_textActions.getActionOrder());
+        _actions = new ArrayList<ActionItem>();
+
+        for (String key: _keys) {
+            _actions.add(actionMap.get(key));
+        }
+    }
+
+    private class Adapter extends RecyclerView.Adapter<Holder> {
+        private List<ActionItem> _actions;
+        ArrayList<Integer> order;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use explicit scoping instead of the default package private level](https://app.codacy.com/manual/gsantner/markor/pullRequest?prid=5344500)